plugins/kubernetes/app/assets/javascripts/kubernetes/controllers/kubernetes_dashboard_ctrl.js

```diff
@@ -1,15 +1,184 @@
-samson.controller('KubernetesDashboardCtrl', function($scope, $stateParams, SseFactory) {
-  $scope.project_id = $stateParams.project_id;
-  $scope.messages = [];
-
-  function init() {
-    // Subscribe to the SSE channel for K8s
-    SseFactory.on('k8s', function(msg) {
-      console.log("Got SSE msg: ", msg);
-      $scope.messages.push(JSON.stringify(msg));
-      _.defer(function() { $scope.$apply(); });
-    });
-  }
-
-  init();
-});
+samson.controller('KubernetesDashboardCtrl',
+  function($rootScope, $scope, $stateParams, $q, environmentsService, kubernetesService, SseFactory, notificationService) {
+    $scope.project_id = $stateParams.project_id;
+    $scope.environments = [];
+    $scope.messages = [];
+    $scope.dashboard_data = [];
+    $scope.environment = undefined;
+    $scope.loadingEnvironments = false;
+    $scope.loadingClusterState = false;
+
+    $scope.reloadDashboardData = function() {
+      loadClusterState();
+    };
+
+    $scope.environmentChanged = function(env) {
+      $scope.environment = env;
+      loadClusterState();
+    };
+
+    $scope.dashboardReady = function() {
+      return !$scope.loadingEnvironments && !$scope.loadingClusterState;
+    };
+
+    $scope.empty = function(collection) {
+      return _.isUndefinedOrEmpty(collection);
+    };
+
+    $scope.notEmpty = function(collection) {
```

Comment:
I tend to use this instead of the negate operator. IMHO, it helps readability, specially when we're mixing Javascript code  with HTML: `<div ng-if="notEmpty(dashboard_data)" ...>` vs `<div ng-if="!empty(dashboard_data)" ...>`

I know it's 3 extra lines of code, but if you guys think there's no gain, I can take it out.